src/helpers/OAuthClient.js

```diff
@@ -0,0 +1,54 @@
+const got = require('got');
+const Cache = require('ttl');
+const qs = require('querystring');
+
+const cache = new Cache({
+  ttl: 3600 * 1000
+});
+if (process.env.NODE_ENV !== 'production') {
+  cache.on('hit', (key, val) => {
+    console.log(`Cache hit for key: ${key}; Value: ${val}`);
+  });
+  cache.on('miss', (key) => {
+    console.log(`Cache miss for key: ${key}`);
+  });
+  cache.on('put', (key, val, ttl) => {
+    console.log(`Cache put for key: ${key}; Value: ${val} with ttl: ${ttl}`);
+  });
+}
+
+async function getAccessToken(authSettings) {
+  let token = null;
+  if (authSettings.oauthConfig) {
+    const { tokenUrl, clientId, clientSecret } = authSettings.oauthConfig;
+    token = cache.get(clientId);
+    if (!token) {
+      const payload = { grant_type: 'client_credentials', client_id: clientId };
```

Comment:
If the client_id is being sent in the header, it's typically not sent in the body.